docs/sources/flow/reference/components/loki.source.heroku.md

```diff
@@ -0,0 +1,136 @@
+block and fans out incoming entries to the list of receivers in `forward_to`.
+
+Before using `loki.source.heroku`, Heroku should be configured with the URL where the Agent will be listening. Follow the steps in [Heroku HTTPS Drain docs](https://devcenter.heroku.com/articles/log-drains#https-drains) for using the Heroku CLI with a command like the following:
+
+```
+heroku drains:add [http|https]://HOSTNAME:PORT/heroku/api/v1/drain -a HEROKU_APP_NAME
+```
+
+Multiple `loki.source.heroku` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.source.heroku "LABEL" {
+  listener {
+    address = "LISTEN_ADDRESS"
+    port = PORT
+  }
+  ...
+
+  forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.heroku` supports the following arguments:
+
+Name                     | Type                   | Description          | Default | Required
+------------------------ | ---------------------- | -------------------- | ------- | --------
+`use_incoming_timestamp` | `bool`                 | Whether or not to use the timestamp received from Heroku. | `false` | no
+`labels`                 | `map(string)`          | The labels to associate with each received Heroku record. | `{}`    | no
+`forward_to`             | `list(LogsReceiver)`   | List of receivers to send log entries to.                 |         | yes
+`relabel_rules`          | `RelabelRules`         | Relabeling rules to apply on log entries.                 | `{}`    | no
+
+The `relabel_rules` field can make use of the `rules` export value from a
+`loki.relabel` component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+## Blocks
+
+The following blocks are supported inside the definition of `loki.source.heroku`:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+listener | [listener] | Configures a listener for Heroku messages. | yes
+
+[listener]: #listener-block
+
+### listener block
+
+The `listener` block defines the listen address and port where the listener
+expects Heroku messages to be sent to.
+
+Name                     | Type          | Description | Default | Required
+------------------------ | ------------- | ----------- | ------- | --------
+`address`                | `string`      | The `<host>` address to listen to for heroku messages. | `0.0.0.0` | no
+`port`                   | `int`         | The `<port>` to listen to for heroku messages. | | yes
+
+Example Block
+
+```river
+listener {
+  address  = "127.0.0.1"
+  port = 8080
+}
+```
+
+## Labels
+
+The `labels` map is applied to every message that the component reads.
+
+The following internal labels are available for relabeling:
+- `__heroku_drain_host`
+- `__heroku_drain_app`
+- `__heroku_drain_proc`
+- `__heroku_drain_log_id`
+
+All url query params will be translated to `__heroku_drain_param_<name>`
```

Comment:
Can we add a note saying that these labels that start with a double underscore are treated as internal and will be discarded unless a relabeling rule is applied to keep them? Similar to how [loki.source.gelf](https://grafana.com/docs/agent/next/flow/reference/components/loki.source.gelf/#:~:text=These%20labels%20are%20stripped%20unless) notes it.